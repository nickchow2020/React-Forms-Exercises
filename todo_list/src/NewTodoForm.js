import React,{useState} from "react"
import "./NewTodoForm.css"

const NewTodoForm = ({addTodo})=> {
    const INITIAL_VALUE = ""
    const [todo,setTodo] = useState(INITIAL_VALUE)
    const handleChange = (e)=>{
        setTodo(
            e.target.value
        )
    }

    const handleSubmit = (e)=>{
        e.preventDefault()
        addTodo(todo)
        setTodo(INITIAL_VALUE)
    }

    return (
        <form onSubmit={handleSubmit}>
            <h1>Add New Todo</h1>
            <label>
                <input 
                name="todo" 
                id="todo" 
                placeholder="add todo" 
                value={todo}
                data-testid="todo"
                onChange={handleChange}
                />
            </label>

            <button data-testid="addTodo">Add Todo</button>
        </form>
    )
}

export default NewTodoForm