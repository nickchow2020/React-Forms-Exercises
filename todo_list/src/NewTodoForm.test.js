import { getByTestId, render } from '@testing-library/react';
import NewTodoForm from './NewTodoForm';

test("should it render",()=>{
    render(<NewTodoForm />)
})

test("should it match with snapshot",()=>{
    const {asFragment} = render(<NewTodoForm />)
    expect(asFragment()).toMatchSnapshot()
})

test("should it having content add new form",()=>{
    const {getByText,getByTestId} = render(<NewTodoForm />)

    expect(getByText("Add New Todo")).toBeInTheDocument()
    expect(getByTestId("addTodo")).toBeInTheDocument()
})