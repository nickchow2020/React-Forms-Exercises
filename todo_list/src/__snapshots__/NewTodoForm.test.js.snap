// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should it match with snapshot 1`] = `
<DocumentFragment>
  <form>
    <h1>
      Add New Todo
    </h1>
    <label>
      <input
        data-testid="todo"
        id="todo"
        name="todo"
        placeholder="add todo"
        value=""
      />
    </label>
    <button
      data-testid="addTodo"
    >
      Add Todo
    </button>
  </form>
</DocumentFragment>
`;
